### PR TITLE
Support installing tools via homebrew

### PR DESCRIPTION
**Pull request by mraerino**
Adding support for Homebrew aka Linuxbrew: https://docs.brew.sh/Homebrew-on-Linux

This shall be amazing because:
- It supports installing A LOT of tools without sudo: https://formulae.brew.sh/formula-linux/
- People can use a `Brewfile` in their git to specify dependencies

I also hope this will allow us to remove some lesser-used tooling from the image and offer users to install it via Homebrew instead.
Good candidates for that:
- pandoc
- hugo
- doxygen
- jq
- optipng
- asciidoctor
- emacs
- gifsicle
- graphicsmagick
- graphviz
- imagemagick
- jpegoptim
- pngcrush
- rsync
- sqlite3
- swig

## Test plan

Has been verified locally with `Brewfile` and custom Brewfile name via `$HOMEBREW_BUNDLE_FILE`

### CODE REVIEW

**Comment by mheffner** [comment]
Nice, I hadn't heard of Homebrew for Linux before. :boom:

**Comment by mraerino** [comment]
yup, i remembered that a german webhosting company offered it along with full-fledged ssh access on a shared host and how this is exactly what we need (hope you can google-translate it): https://blog.uberspace.de/brauen-oder-toasten/

**Comment by mraerino** [comment]
this is now based on an older commit, because the swift commit fails all builds

**Comment by mraerino** [comment]
This is an example build which uses this feature successfully: https://app.netlify.com/sites/edge-functions-ts/deploys/5eb20aaf6ddfc20006027e9d

**Comment by mheffner** [comment]
What if we removed this from the included_software doc so that we made it clear it's not generally available, then we could continue testing without officially "supporting" it? Then we could decide on a release date if this is a pattern we like.

**Comment by calavera** [comment]
> What if we removed this from the included_software doc so that we made it clear it's not generally available, then we could continue testing without officially "supporting" it? Then we could decide on a release date if this is a pattern we like.

I think that it's a good idea, or even mark it as experimental. This change as it is makes much more sense than other tools that we've merged before, and I don't think we should hold it back.

**Comment by mraerino** [comment]
this is currently blocked on https://github.com/netlify/build-image/pull/401
once that is merged i'm happy to rebase and update notes

**Review by calavera — APPROVED** [review]

**Comment by mraerino** [comment]
@netlify/backend are you happy to merge this change?

File: run-build-functions.sh
Comment (mheffner) [inline run-build-functions.sh:454]: fwiw, `-n` should also be equivalent to `! -z`

File: Dockerfile
Comment (mheffner) [inline Dockerfile:491]: Any concern that we aren't pulling from a static/stable tag?

**Review by mheffner — APPROVED** [review]
Yeah, let's get it out

File: Dockerfile
Comment (mraerino) [reply Dockerfile:491]: that's a good point. let's do that as part of a general release

**Pull request merged**
